### DISCUSSION TITLE
Passlib dependency still advisable?

### DISCUSSION THREAD

**Discussion by SpoonOfDoom** [fastapi/full-stack-fastapi-template#1369] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
not applicable
```

### Description

It appears to be that passlib is abandoned. It has been for a while, and when looming problems (e.g. with Python 3.13) became more apparent, the maintainer made some comments and promised to take up development again, but that has been 6 months ago he has been unresponsive again since then, and hasn't given anyone access to pick up the mantle. It appears we may not be getting regular maintanence in the future (see the discussion in the passlib repo for context: https://foss.heptapod.net/python-libs/passlib/-/issues/187).

Now that raises two questions for this project template, which I think are worth at least discussing since it's a security related package.

1. Should this project still use passlib as a dependency (and thus "promote" the use of it)?
2. If not, what would be the best replacement?

Unless any new vulnerabilities pop up, it should be fine for current Python versions, but could become a problem in the near-ish future. It'd also be a problem if any new vulnerabilities that affect passlib were to pop up.

### Operating System

Other

### Operating System Details

Not relevant

### Python Version

3.11.9

**Comment by storenth** [comment]
No, looks like community suggest to use `bcrypt` directly.
Guys, please remove `passlib`
 https://github.com/fastapi/full-stack-fastapi-template/blob/d3d370cad04228a9e5a4a374a007c0f4f9f068a7/backend/pyproject.toml#L10
 due to project dropped and no more support by maintainer.
https://passlib.readthedocs.io/en/latest/install.html

Related issue/discuss: https://github.com/pyca/bcrypt/issues/684

**Comment by YuriiMotov** [comment]
There is a PR: https://github.com/fastapi/full-stack-fastapi-template/pull/1539

**Comment by tiangolo** [comment, marked as answer]
This was handled here: https://github.com/fastapi/full-stack-fastapi-template/pull/2104 :coffee:
